src/essreflectometry/orso.py

```diff
@@ -1,15 +1,101 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
-import warnings
+"""ORSO utilities for reflectometry."""
+import platform
+from datetime import datetime, timezone
+from typing import NewType, Optional
 
+import sciline
+from orsopy.fileio import base as orso_base
+from orsopy.fileio import data_source, reduction
 
-def not_found_warning():
+from . import __version__
+from .types import Reference, Run, Sample
+
+
+class OrsoExperiment(
+    sciline.Scope[Run, data_source.Experiment], data_source.Experiment
+):
+    """ORSO experiment for a run."""
+
+
+class OrsoInstrument(
+    sciline.Scope[Run, data_source.InstrumentSettings], data_source.InstrumentSettings
+):
+    """ORSO instrument settings for a run."""
+
+
+class OrsoOwner(sciline.Scope[Run, orso_base.Person], orso_base.Person):
+    """ORSO owner of a file."""
+
+
+class OrsoReduction(sciline.Scope[Run, reduction.Reduction], reduction.Reduction):
+    """ORSO measurement for a run."""
+
+
+class OrsoSample(sciline.Scope[Run, data_source.Sample], data_source.Sample):
+    """ORSO sample of a run."""
```

Comment:
Do we need to support all this for the `Reference` measurement? Or is it recording only for the sample run (i.e., can we use `NewType` instead of a "generic"?

Reply:
Depends on whether we want to process a file in isolation and store it as ort.

Reply:
Looking at this again, I still do not see the case for parametrizing all of the above with the run type. I feel it would be simpler and clear to just use the sample everywhere.

- The sample and reference are part of the same experiment.
- The same instrument is used, it is always a reference measurement at that instrument.
- No one cares about the owner of the reference file, I'd say. I think typically it is done by the same user during their experiment.
- There is just one reduction, not a reduction for the sample and one for the reference (or if there is, the reference run would be the sample run?).